ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/websocket/subscription/blockheaders/NewBlockHeadersSubscriptionService.java

```diff
@@ -38,20 +38,28 @@ public NewBlockHeadersSubscriptionService(
 
   @Override
   public void onBlockAdded(final BlockAddedEvent event, final Blockchain blockchain) {
-    final List<NewBlockHeadersSubscription> subscribers =
-        subscriptionManager.subscriptionsOfType(
-            SubscriptionType.NEW_BLOCK_HEADERS, NewBlockHeadersSubscription.class);
+    subscriptionManager
+        .getVertx()
+        .<Void>executeBlocking(
```

Comment:
It would probably be better if instead of `SubscriptionManager` exposing a `getVertx` and then having these classes be directly dependent on vertx,  just expose a method to jump to a worker thread.  That could be a simple `executeOnWorkerThread` that takes a runnable or a really nice API would be to combine the jump to a different thread with looking up the subscribers.  So:
```
subscriptionManager.notifySubscribers(SubscriptionType.NEW_BLOCK_HEADERS, NewBlockHeadersSubscription.class, subscribers -> {
   final Hash new BlockHash = event.getBlock().getHash()
   ...
});
```

That would also avoid having to duplicate this code to jump to another thread for every subscription service.